src/registrar/views/index.py

```diff
@@ -1,21 +1,46 @@
 from django.shortcuts import render
 
 from registrar.models import DomainApplication, Domain, UserDomainRole
+from registrar.models.draft_domain import DraftDomain
 
 
 def index(request):
     """This page is available to anyone without logging in."""
     context = {}
     if request.user.is_authenticated:
-        applications = DomainApplication.objects.filter(creator=request.user)
         # Let's exclude the approved applications since our
         # domain_applications context will be used to populate
         # the active applications table
-        context["domain_applications"] = applications.exclude(status="approved")
+        applications = DomainApplication.objects.filter(creator=request.user).exclude(status="approved")
+
+        # Adds display logic for empty domain requests
+        counter = 1
+        for application in applications:
+            if not application.requested_domain or not application.requested_domain.name:
+                application.requested_domain = DraftDomain(name=f"New domain request {counter}")
+                counter += 1
```

Comment:
**Question:** Is it worth testing this scenario of "user exits without putting in a URL yet and want to make sure naming shows up correctly"? Don't know if it's worth (I also don't see in the Figma a sketch), but thought it'd be worth a mention

Reply:
Yeah that's a good point. I am going to refactor how this bit works so I'll include that there